ts/main.ts

```diff
@@ -259,24 +279,128 @@ function renderDetails(anime: Search): HTMLDivElement {
   return $row;
 }
 
+function renderWatchlist(entry: Search): HTMLLIElement {
+  const $listItem = document.createElement('li');
+  $listItem.setAttribute('data-id', String(entry.animeId));
+  const $row = document.createElement('div');
+  $row.setAttribute('class', 'row search-row');
+  const $columnHalf1 = document.createElement('div');
+  $columnHalf1.setAttribute('class', 'column-half style-column image-column');
+  const $image = document.createElement('img');
+  $image.setAttribute('src', entry.imageURL);
+  const $columnHalf2 = document.createElement('div');
+  $columnHalf2.setAttribute('class', 'column-half style-column text-column');
+  const $title = document.createElement('h2');
+  $title.setAttribute('class', 'title text');
+  $title.textContent = entry.title;
+  const $episodes = document.createElement('h3');
+  $episodes.setAttribute('class', 'episodes text');
+  $episodes.textContent = `Eps ${entry.episodes}`;
+  const $moreDetails = document.createElement('a');
+  $moreDetails.setAttribute('href', '#');
+  $moreDetails.setAttribute('class', 'details text');
+  $moreDetails.textContent = 'More details...';
+  $moreDetails.addEventListener('click', async (event: Event) => {
+    while ($dataView[2].lastChild) {
+      $dataView[2].removeChild($dataView[2].lastChild);
+    }
+    const $eventTarget = event.target as HTMLElement;
+    const $closest = $eventTarget.closest('[data-id]') as HTMLAnchorElement;
+    const animeId = $closest.getAttribute('data-id');
+    try {
+      const response = await fetch(`https://api.jikan.moe/v4/anime/${animeId}`);
+      if (!response.ok) throw new Error('Network response was not ok');
+      const individualAnime = await response.json();
+      const anime = {
+        title: individualAnime.data.title,
+        imageURL: individualAnime?.data?.images?.jpg?.image_url,
+        episodes: individualAnime.data.episodes,
+        type: individualAnime.data.type,
+        status: individualAnime.data.status,
+        aired: individualAnime.data.aired.string,
+        premiered: individualAnime.data.season,
+        rating: individualAnime.data.rating,
+        synopsis: individualAnime.data.synopsis,
+        animeId: individualAnime.data.mal_id,
+      };
+      const details = renderDetails(anime);
+      $dataView[2].appendChild(details);
+      viewSwap('details');
+    } catch (error) {
+      console.error('There was a problem with your fetch:', error);
+    }
```

Comment:
It might be better to extract this callback function into a global function and pass it in as a reference here. It would make this render function a bit smaller and more manageable to read